Lib/_compression.py

```diff
@@ -110,6 +110,18 @@ def read(self, size=-1):
         self._pos += len(data)
         return data
 
+    def readall(self):
+        chunks = []
+        while True:
+            # sys.maxsize means that the max length of output buffer is
+            # unlimited, so that the whole input buffer can be decompressed
+            # within one .decompress() call.
+            data = self.read(sys.maxsize)
+            if not data:
+                break
+            chunks.append(data)
+        return b''.join(chunks)
```

Comment:
How about this microoptimization?

```py
return b''.join(iter(functools.partial(self.read, sys.maxsize), b''))
```

Reply:
It has improvement, but not very obvious.
```
linux-2.6.39.4.tar.bz2
input size: 76,097,195, output size: 449,638,400
best of 10: [e:\baseline_file] 9.532 sec -> [e:\patched_file] 9.527 sec, 1.00x faster (-0%)

firefox-79.0.linux-i686.tar.bz2
input size: 74,109,706, output size: 228,055,040
best of 10: [e:\baseline_file] 6.718 sec -> [e:\patched_file] 6.712 sec, 1.00x faster (-0%)

ffmpeg-4.3.1.tar.bz2
input size: 11,301,038, output size: 74,567,680
best of 10: [e:\baseline_file] 1.407 sec -> [e:\patched_file] 1.407 sec, 1.00x slower (+0%)

gimp-2.10.20.tar.bz2
input size: 33,108,938, output size: 214,179,840
best of 10: [e:\baseline_file] 4.116 sec -> [e:\patched_file] 4.129 sec, 1.00x slower (+0%)

sde-external-8.56.0-2020-07-05-lin.tar.bz2
input size: 26,746,086, output size: 92,129,280
best of 10: [e:\baseline_file] 2.480 sec -> [e:\patched_file] 2.475 sec, 1.00x faster (-0%)

linux-5.7.10.tar.xz
input size: 112,722,840, output size: 966,062,080
best of 10: [e:\baseline_file] 7.458 sec -> [e:\patched_file] 7.429 sec, 1.00x faster (-0%)

linux-2.6.39.4.tar.xz
input size: 63,243,812, output size: 449,638,400
best of 10: [e:\baseline_file] 4.182 sec -> [e:\patched_file] 4.189 sec, 1.00x slower (+0%)

gcc-9.3.0.tar.xz
input size: 70,533,868, output size: 618,608,640
best of 10: [e:\baseline_file] 4.850 sec -> [e:\patched_file] 4.853 sec, 1.00x slower (+0%)

Python-3.8.5.tar.xz
input size: 18,019,640, output size: 87,531,520
best of 10: [e:\baseline_file] 1.069 sec -> [e:\patched_file] 1.069 sec, 1.00x faster (-0%)

firefox-79.0.source.tar.xz
input size: 333,220,776, output size: 2,240,573,440
best of 10: [e:\baseline_file] 19.832 sec -> [e:\patched_file] 19.748 sec, 1.00x faster (-0%)

linux-5.7.10.tar.gz
input size: 175,493,567, output size: 966,062,080
best of 10: [e:\baseline_file] 3.712 sec -> [e:\patched_file] 3.597 sec, 1.03x faster (-3%)

linux-2.6.39.4.tar.gz
input size: 96,011,469, output size: 449,638,400
best of 10: [e:\baseline_file] 1.786 sec -> [e:\patched_file] 1.781 sec, 1.00x faster (-0%)

gcc-9.3.0.tar.gz
input size: 124,140,238, output size: 618,608,640
best of 10: [e:\baseline_file] 2.324 sec -> [e:\patched_file] 2.307 sec, 1.01x faster (-1%)

Python-3.8.5.tgz
input size: 24,149,103, output size: 87,531,520
best of 10: [e:\baseline_file] 0.334 sec -> [e:\patched_file] 0.336 sec, 1.00x slower (+0%)

openjdk-14.0.2_linux-x64_bin.tar.gz
input size: 198,606,200, output size: 335,175,680
best of 10: [e:\baseline_file] 1.681 sec -> [e:\patched_file] 1.691 sec, 1.01x slower (+1%)

postgresql-10.12-1-linux-x64-binaries.tar.gz
input size: 159,090,047, output size: 408,678,400
best of 10: [e:\baseline_file] 1.952 sec -> [e:\patched_file] 1.940 sec, 1.01x faster (-1%)

```

Reply:
I tested twice, only this file 1.03x faster, the others are not obvious, don't know why.

```
linux-5.7.10.tar.gz
input size: 175,493,567, output size: 966,062,080
best of 10: [e:\baseline_file] 3.704 sec -> [e:\patched_file] 3.590 sec, 1.03x faster (-3%)
```